pdata/internal/cmd/pdatagen/internal/immutable_slice_structs.go

```diff
@@ -0,0 +1,225 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "go.opentelemetry.io/collector/pdata/internal/cmd/pdatagen/internal"
+
+import (
+	"os"
+	"strings"
+)
+
+const immutableSliceTemplate = `// ${structName} represents a []${itemType} slice that cannot be mutated.
```

Comment:
Not sure if it is worth to generate these slices. I will give some attention to this generator while you are away.